src/config.rs

```diff
@@ -334,6 +334,83 @@ impl Context {
     }
 }
 
+// Separate impl block for self address handling
+impl Context {
+    /// determine whether the specified addr maps to the/a self addr
+    pub(crate) async fn is_self_addr(&self, addr: &str) -> Result<bool> {
+        let addr = addr_normalize(addr).to_lowercase();
+
+        // The addresses we get here are already normalized and lowercase
+        Ok(
+            self.get_config(Config::ConfiguredAddr).await?.as_deref() == Some(&addr)
+                || self.get_secondary_self_addrs().await?.contains(&addr),
+        )
+    }
+
+    /// Sets `primary_new` as the new primary self address and saves the old
+    /// primary address (if exists) as a secondary address.
+    pub(crate) async fn set_primary_self_addr(&self, primary_new: &str) -> Result<()> {
+        let primary_new = addr_normalize(primary_new).to_lowercase();
+
+        // add old primary address (if exists) to secondary addresses
+        let mut secondary_addrs = self.get_all_self_addrs().await?;
+
+        for a in secondary_addrs.iter_mut() {
+            *a = addr_normalize(a).to_lowercase();
+        }
+
+        // never store a primary address also as a secondary
+        secondary_addrs.retain(|a| a != &primary_new);
+        self.set_config(
+            Config::SecondaryAddrs,
+            Some(secondary_addrs.join(" ").as_str()),
+        )
+        .await?;
+
+        self.set_config(Config::ConfiguredAddr, Some(&primary_new))
+            .await?;
+
+        Ok(())
+    }
+
+    /// Returns all primary and secondary self addresses.
+    pub(crate) async fn get_all_self_addrs(&self) -> Result<Vec<String>> {
+        let mut ret = Vec::new();
+
+        ret.push(self.get_primary_self_addr().await.unwrap_or_default());
+        ret.extend(self.get_secondary_self_addrs().await?.into_iter());
+
+        Ok(ret)
+    }
+
+    /// Returns all secondary self addresses.
+    ///
+    /// The addresses are already normalized and lowercased in the database.
+    pub(crate) async fn get_secondary_self_addrs(&self) -> Result<Vec<String>> {
+        let secondary_addrs = self
+            .get_config(Config::SecondaryAddrs)
+            .await?
+            .unwrap_or_default();
+        Ok(secondary_addrs
+            .split_ascii_whitespace()
+            .map(|s| s.to_string())
+            .collect())
+    }
+
+    /// Returns the primary self address.
+    ///
+    /// Normalizes and lowercases the address since the ConfiguredAddr
+    /// may not be lowercased (to start lowercasing it now, we would
+    /// need a db migration - which we can do at some point in the future)
+    pub async fn get_primary_self_addr(&self) -> Result<String> {
+        let ret = self
+            .get_config(Config::ConfiguredAddr)
+            .await?
+            .context("No self addr configured")?;
+        Ok(addr_normalize(&ret).to_lowercase())
+    }
```

Comment:
Not sure if this "normalizes and lowercases" is the correct approach.

Before this PR, `is_self_addr()` used `addr_cmp()` to do a case-insensitive comparison. Apart from this, case was preserved.

It seems easier to me to just lowercase stuff, and I have never seen an email address where it was important that it stays uppercase - but maybe it exists?

If we say that case is important to preserve (while I'm pretty sure that it's important to compare case-insensitive in case the user's keyboard auto-capitalizes the email addr), we probably have to make sure that:
- the sql `NOT IN` statements are case-insensitive - does anyone know if they are?
- `set_primary_self_addr()` and `is_self_addr()` compare case-insensitively

Reply:
i think, it is better to stay with preserving case and ignoring case on comparison.

this is what other MUA do as well (i just tested thunderbird) - and also Delta Chat preserves case up to now. i think, ppl expect addresses to be shown as entered to other ppl, Prename.Name@domain looks better for some ppl than prename.name@domain.

Reply:
I pushed 250e20e37922a01ed772598ab40178ec8c013452. And I just checked that the sql `NOT IN` statements are case-insensitive (because of the `COLLATE NOCASE`).